src/libAtomVM/globalcontext.c

```diff
@@ -552,6 +552,23 @@ int globalcontext_get_registered_process(GlobalContext *glb, int atom_index)
     return 0;
 }
 
+term globalcontext_get_registered_name_process(GlobalContext *glb, int local_process_id)
+{
+    struct ListHead *registered_processes_list = synclist_wrlock(&glb->registered_processes);
+    struct ListHead *item;
+    struct ListHead *tmp;
+    MUTABLE_LIST_FOR_EACH (item, tmp, registered_processes_list) {
+        struct RegisteredProcess *registered_process = GET_LIST_ENTRY(item, struct RegisteredProcess, registered_processes_list_head);
```

Comment:
@petermm is there a reason we need write lock and `MUTABLE_LIST_FOR_EACH` here? I wonder if it's a leftover from prev impl or I missed something.

Reply:
Why wouldn't we need a lock here?

Reply:
Sorry, I was thinking about read lock vs read-write lock. We don't modify registered processes here.

Reply:
Good question, believe I just copied over some implementation, so maybe it is fine with rdlock - no strong opinion here. @pguyot will know for sure.